### ISSUE TITLE
Dependencies on podspec not integrated to the project

### ISSUE THREAD

**Issue by Vektrat** [CocoaPods/CocoaPods#5155]
# Report

## What did you do?
I'm trying to create a new pod library, and I've read the documentation a hundred times, but I cannot seem to get past this struggle, so sorry for opening an issue here but I've also digged on StackOverflow.

I'm not even sure if what I'm trying to accomplish is supported - I'm trying to reference other pod dependencies in the podspec file (in the **s.dependency**-es section), which are needed in the Classes I'm providing in my recently created pod library. However, the code fails to find them in the project. If I try to add them as frameworks, they are not in the list, and if I try to import them, they aren't found either. I believe if I just copy the files into the Classes directory it will work, but I wanted to have a nicely built pod project. In the built log, this is what I get:
```
pod lib create MyProject
- Configured it and added files
pod spec lint MyProject.podspec --verbose
```

```
...
Resolving dependencies of 
Comparing resolved specification to the sandbox manifest
  A Alamofire
  A MyProject
  A ReachabilitySwift
  A SwiftyJSON

Downloading dependencies
-> Installing Alamofire (3.3.1)
  > Copying Alamofire from `/Users/vektrat/Library/Caches/CocoaPods/Pods/Release/Alamofire/3.3.1-369bc` to
  `../../../../private/var/folders/n2/_wn23wqs5w12l6yycdk11pyc0000gn/T/CocoaPods/Lint/Pods/Alamofire`

-> Installing MyProject (0.1.0)
  > Copying MyProject from `/Users/vektrat/Library/Caches/CocoaPods/Pods/External/MedtepCore/1395d33878e44ec653fa57cb86526f28-71259` to
  `../../../../private/var/folders/n2/_wn23wqs5w12l6yycdk11pyc0000gn/T/CocoaPods/Lint/Pods/MyProject`

-> Installing ReachabilitySwift (2.3.3)
  > Copying ReachabilitySwift from `/Users/vektrat/Library/Caches/CocoaPods/Pods/Release/ReachabilitySwift/2.3.3-e0f6b` to
  `../../../../private/var/folders/n2/_wn23wqs5w12l6yycdk11pyc0000gn/T/CocoaPods/Lint/Pods/ReachabilitySwift`

-> Installing SwiftyJSON (2.3.2)
  > Copying SwiftyJSON from `/Users/vektrat/Library/Caches/CocoaPods/Pods/Release/SwiftyJSON/2.3.2-04cce` to
  `../../../../private/var/folders/n2/_wn23wqs5w12l6yycdk11pyc0000gn/T/CocoaPods/Lint/Pods/SwiftyJSON`
  - Running pre install hooks

Generating Pods project
  - Creating Pods project
  - Adding source files to Pods project
  - Adding frameworks to Pods project
  - Adding libraries to Pods project
  - Adding resources to Pods project
...
```

## What did you expected to happen?

To validate correctly

## What happened instead?

Failed with errors everywhere it referenced 3rd party libs with (error: use of undeclared type)

## Podspec
I believe, though I may be wrong, that the podfile is needed just for the tests which I'm not doing.
This is a snippet of my podspec

```
  # s.public_header_files = 'Pod/Classes/**/*.h'
  # s.frameworks = 'UIKit', 'MapKit'
  s.dependency 'Alamofire', '~> 3.3'
  s.dependency 'SwiftyJSON', '~> 2.3'
  s.dependency 'ReachabilitySwift', '~> 2.3'
```

I get the feeling either I'm trying something that cannot be done, or that I am mixing pod concepts here.

Thank you for your time

**Comment by segiddins** [comment]
Are you `import`ing the dependencies in the files you're using them in?

**Comment by Vektrat** [comment]
Hello, thanks for the quick reply.

I've tried so yes, but I immediately get an xcode error saying : **No such module 'XXXX**

**Comment by segiddins** [comment]
In that case, can you share a project we can use to test against? It's kinda hard to guess whats going wrong without seeing concrete errors.

**Comment by Vektrat** [comment]
I believe I got it working, I'll describe what I think was the problem, however I can't guarantee this was the exact cause since I also ran into some compatibility codes along the development (with pods and ruby versions).

Since I was starting the pod and basically testing the library I kept reusing the git tag 0.1.0, and I assumed the validator checked my changes locally. However, it seems every time I lint-ed it was done against the remote copy, which cached, and probably didn't update correctly due to the tag re-using. So by pushing all the changes again, deleting the tag and creating it again, and cleaning the pods cache for the new pod, changes are accounted for and, even though Xcode sometimes still marks my imports as missing, when validating it is working.

Thanks for your time and suggestions segiddins.
